Source/ACE.Server/WorldObjects/Player_Commerce.cs

```diff
@@ -242,20 +242,16 @@ public void HandleActionSellItem(List<ItemProfile> itemprofiles, ObjectGuid vend
                 else
                 {
                     // remove item from inventory.
-                    TryRemoveFromInventory(profile.Guid);
+                    TryRemoveFromInventoryWithNetworking(item);
                 }
 
                 //Session.Network.EnqueueSend(new GameMessagePrivateUpdateInstanceId(profile, PropertyInstanceId.Container, new ObjectGuid(0).Full));
 
                 item.SetPropertiesForVendor();
-
-                // clean up the shard database.
-                throw new NotImplementedException();
-                // todo fix for EF
-                //DatabaseManager.Shard.DeleteObject(item.SnapShotOfAceObject(), null);
-
                 Session.Network.EnqueueSend(new GameMessageDeleteObject(item));
                 purchaselist.Add(item);
+                //update item in database
+                item.SaveBiotaToDatabase();
```

Comment:
Can you add the comments here similar to the comments above the other item.SaveBiotaToDatabase in player_inventory

In the future, i'm HOPING to add an improved way to save just the parent property id for these items which should be much smaller hits on the db than saving the whole object like SaveBiotaToDatabase() does currently.

I think my note basically talked about that.